### ISSUE TITLE
Webpack devServer - separating CSS and JS from the rest of files

### ISSUE THREAD

**Issue by bencergazda** [webpack/webpack-dev-server#1237]
* Operating System: Windows 10
* Node Version: 8.9.1
* NPM Version: 5.5.1
* webpack Version: 3.10.0
* webpack-dev-server Version: 2.9.5

- [ ] This is a **bug**
- [x] This is a **feature** request
- [ ] This is a **modification** request

### Code

```
// project dir structure
/dev
    /assets
        [all imported static assets, like WOFFs, SVGs, PNGs etc...]
    /css
        styles.css
    /js
        bundle.js

/src
    /sass
        styles.scss
    /js
        scripts.js

index.html
```

```js
// webpack.config.js
{
    context: './src/',
    entry: {
        'js/bundle.js': './js/scripts.js',
        'css/styles.css': './sass/styles.scss'
    },
    output: {
        filename: '../[filename]',
        path: './',
        publicPath: '/dev/assets/'
    },

    plugins: [
        new ExtractTextPlugin('../[name]')
    ],
    ...
    devServer: {
        contentBase: './',
        publicPath: '/dev/assets'
        ...
    }
}
```

### Expected Behavior
The concept here is that every file will be placed to the `dev/assets` folder by default, but in case of the entry files, I have the option to create them to other folders (under `dev`), described by their entry object key (the trick is the `../[filename]` value of the filename property).

### Actual Behavior
This looks to be a valid webpack configuration and indeed works as expected with normal Webpack builds, but the same configuration will not work with Webpack dev server, as it will only serve the files from `dev/assets`, and the files of it's sibling folders (like `dev/js`) cannot be accessed.

### For Features; What is the motivation and/or use-case for the feature?
This would give me full control over the folder and name structure of the `dev` files, and also collect the rest of the files (static assets like font files, SVGs, PNGs etc) to the `dev/assets` folder. This would also make me possible to use the same directory tree in dist mode, without the `dev` folder

```
// project dir structure
/assets (dist)
    [all imported static assets, like WOFFs, SVGs, PNGs etc...]
/css (dist)
    styles.css
/dev
    /assets
        [all imported static assets, like WOFFs, SVGs, PNGs etc...]
    /css
        styles.css
    /js
        bundle.js
/js (dist)
    bundle.js

/src
    /sass
        styles.scss
    /js
        scripts.js

index.html
```

**Comment by shellscape** [comment]
@bencergazda thanks for checking in. We'd be happy to review a pull request for this feature!

**Comment by michael-ciniawsky** [comment]
```diff
    devServer: {
        contentBase: './',
-       publicPath: '/dev/assets'
+       publicPath: '/dev/
        ...
    }
```

**Comment by bencergazda** [comment]
@michael-ciniawsky Thanks for your response. The reason why I used `/dev/assets` as publicPath was to automatically collect all asset (not entry) files to an `assets` folder. (The entry files files got a `../` prefix, see in `output.filename` to pretend a relativity to `/dev` instead of `/dev/assets`).
But after some investigation I now realized that I can set a different `outputPath` also for `url-loader` and `file-loader` in the specific `module.rules`, where I generate the asset files, so I could remove my ugly hack from the webpack config.
